Signal/src/ViewControllers/ConversationView/Cells/OWSContactOffersCell.m

```diff
@@ -126,67 +159,36 @@ - (CGFloat)buttonVPadding
     return 5.f;
 }
 
-- (CGFloat)buttonVSpacing
+- (CGFloat)vSpacing
 {
     return 5.f;
 }
 
-- (void)layoutSubviews
+- (CGFloat)buttonHeight
 {
-    [super layoutSubviews];
-
-    OWSContactOffersInteraction *interaction = (OWSContactOffersInteraction *)self.viewItem.interaction;
-
-    // We're using a bit of a hack to get this and the unread indicator to layout as
-    // "full width" cells.  These cells will end up with an erroneous left margin that we
-    // want to reverse.
-    CGFloat contentWidth = self.width;
-    CGFloat left = -self.left;
-
-    CGRect titleFrame = self.contentView.bounds;
-    titleFrame.origin = CGPointMake(left + self.hMargin, self.topVMargin);
-    titleFrame.size.width = contentWidth - 2 * self.hMargin;
-    titleFrame.size.height = ceil([self.titleLabel sizeThatFits:CGSizeZero].height);
-    self.titleLabel.frame = titleFrame;
-
-    __block CGFloat y = round(self.titleLabel.bottom + self.buttonVSpacing);
-    void (^layoutButton)(UIButton *, BOOL) = ^(UIButton *button, BOOL isVisible) {
-        if (isVisible) {
-            button.hidden = NO;
-
-            button.frame = CGRectMake(round(left + self.hMargin),
-                round(y),
-                floor(contentWidth - 2 * self.hMargin),
-                ceil([button sizeThatFits:CGSizeZero].height + self.buttonVPadding));
-            y = round(button.bottom + self.buttonVSpacing);
-        } else {
-            button.hidden = YES;
-        }
-    };
-
-    layoutButton(self.addToContactsButton, interaction.hasAddToContactsOffer);
-    layoutButton(self.addToProfileWhitelistButton, interaction.hasAddToProfileWhitelistOffer);
-    layoutButton(self.blockButton, interaction.hasBlockOffer);
+    return (self.buttonVPadding + CGSizeCeil([self.addToContactsButton sizeThatFits:CGSizeZero]).height);
 }
 
-- (CGSize)cellSizeForViewWidth:(int)viewWidth contentWidth:(int)contentWidth
+- (CGSize)cellSize
```

Comment:
What is the motivation for removing the parameter and instead keeping layout info statefully on this model?

Reply:
See above - there's now a ton of "constants" and "semi-constants" the are calculated from each other.  I'm trying to avoid: 

* Passing them all around separately
* Calculating them all for each cell.

We could maintain an instance of "conversation layout info/style/appearance" and pass it into the cell's "configure" and "measurement" methods?  Would that feel better to you?  If it's easy, I'd be happy to do it.